<a href="https://colab.research.google.com/github/williammcintosh/CS510_DataEngineering_Assignments/blob/main/07_integrartion/07_Integration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Week 07 Integration

## Will McIntosh

# A. Discussion Question



In [1]:
%%capture
! gdown 18UD2EGKwq3YjVMBPsS8kVz0e671p7vzV
! gdown 1xAWZ_tEjEAFsosCdax78lrWkLnUgoWSn

In [2]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

pd.set_option('display.max_colwidth', None)
pd.set_option('display.max_columns', None)

In [3]:
covid_df = pd.read_csv("/content/COVID_county_data.csv")
census_df = pd.read_csv("/content/acs2017_census_tract_data.csv")

In [4]:
covid_df.head()

date     county       state     fips  cases  deaths
0  2020-01-21  Snohomish  Washington  53061.0      1     0.0
1  2020-01-22  Snohomish  Washington  53061.0      1     0.0
2  2020-01-23  Snohomish  Washington  53061.0      1     0.0
3  2020-01-24       Cook    Illinois  17031.0      1     0.0
4  2020-01-24  Snohomish  Washington  53061.0      1     0.0

In [5]:
census_df.head()

TractId    State          County  TotalPop   Men  Women  Hispanic  \
0  1001020100  Alabama  Autauga County      1845   899    946       2.4   
1  1001020200  Alabama  Autauga County      2172  1167   1005       1.1   
2  1001020300  Alabama  Autauga County      3385  1533   1852       8.0   
3  1001020400  Alabama  Autauga County      4267  2001   2266       9.6   
4  1001020500  Alabama  Autauga County      9965  5054   4911       0.9   

   White  Black  Native  Asian  Pacific  VotingAgeCitizen   Income  IncomeErr  \
0   86.3    5.2     0.0    1.2      0.0              1407  67826.0    14560.0   
1   41.6   54.5     0.0    1.0      0.0              1652  41287.0     3819.0   
2   61.4   26.5     0.6    0.7      0.4              2480  46806.0     9496.0   
3   80.3    7.1     0.5    0.2      0.0              3257  55895.0     4369.0   
4   77.5   16.4     0.0    3.1      0.0              7229  68143.0    14424.0   

   IncomePerCap  IncomePerCapErr  Poverty  ChildPoverty  Professional  \
0       33018.0           6294.0     10.7          20.8          38.5   
1       18996.0           2453.0     22.4          35.8          30.5   
2       21236.0           2562.0     14.7          21.1          27.9   
3       28068.0           3190.0      2.3           1.7          29.0   
4       36905.0          10706.0     12.2          17.9          48.8   

   Service  Office  Construction  Production  Drive  Carpool  Transit  Walk  \
0     15.6    22.8          10.8        12.4   94.2      3.3      0.0   0.5   
1     24.9    22.9           6.3        15.4   90.5      9.1      0.0   0.0   
2     19.4    33.3           9.9         9.6   88.3      8.4      0.0   1.0   
3     16.6    25.8           9.1        19.5   82.3     11.2      0.0   1.5   
4     13.8    20.5           3.5        13.4   86.9     11.2      0.0   0.8   

   OtherTransp  WorkAtHome  MeanCommute  Employed  PrivateWork  PublicWork  \
0          0.0         2.1         24.5       881         74.2        21.2   
1          0.5         0.0         22.2       852         75.9        15.0   
2          0.8         1.5         23.1      1482         73.3        21.1   
3          2.9         2.1         25.9      1849         75.8        19.7   
4          0.3         0.7         21.0      4787         71.4        24.1   

   SelfEmployed  FamilyWork  Unemployment  
0           4.5         0.0           4.6  
1           9.0         0.0           3.4  
2           4.8         0.7           4.7  
3           4.5         0.0           6.1  
4           4.5         0.0           2.3

In [6]:
# Group by state and county, and aggregate relevant data
county_info = census_df.groupby(['State', 'County']).agg({
    'TotalPop': 'sum',         # Sum total population
    'Poverty': 'mean',         # Average poverty, weighted by population might be more appropriate
    'IncomePerCap': 'mean'     # Average income per capita, weighted by population might be more appropriate
}).reset_index()

# Rename columns to match the requirements
county_info.columns = ['State', 'County', 'Population', '%Poverty', 'PerCapitaIncome']

# Adding a unique ID column for County - this might need to correspond to some identifier in the COVID data
# For illustrative purposes, we'll generate a simple combination of state and county names
county_info['ID'] = county_info['County'] + ', ' + county_info['State']

In [7]:
county_info

State             County  Population   %Poverty  PerCapitaIncome  \
0     Alabama     Autauga County       55036  14.558333     26588.166667   
1     Alabama     Baldwin County      203360  12.874194     29130.709677   
2     Alabama     Barbour County       26201  27.755556     17891.666667   
3     Alabama        Bibb County       22580  13.925000     21799.000000   
4     Alabama      Blount County       57667  16.422222     21598.444444   
...       ...                ...         ...        ...              ...   
3215  Wyoming  Sweetwater County       44527  11.691667     32898.083333   
3216  Wyoming       Teton County       22923   7.550000     47706.250000   
3217  Wyoming       Uinta County       20758  14.733333     27086.000000   
3218  Wyoming    Washakie County        8253  13.066667     26828.333333   
3219  Wyoming      Weston County        7117  14.200000     31058.500000   

                              ID  
0        Autauga County, Alabama  
1        Baldwin County, Alabama  
2        Barbour County, Alabama  
3           Bibb County, Alabama  
4         Blount County, Alabama  
...                          ...  
3215  Sweetwater County, Wyoming  
3216       Teton County, Wyoming  
3217       Uinta County, Wyoming  
3218    Washakie County, Wyoming  
3219      Weston County, Wyoming  

[3220 rows x 6 columns]

# B. Transform the ACS Data

1. Aggregate the census data at the county level.
2. Drop unnecessary columns from the aggregated data.
3. Add an ID column that relates to the COVID data.
4. Create a specific set of information about select counties.
5. Identify the most and least populous counties in the USA.


In [8]:
# Filtering out the data for specified counties
counties = [
    "Loudoun County, Virginia",
    "Washington County, Oregon",
    "Harlan County, Kentucky",
    "Malheur County, Oregon"
]
filtered_counties = county_info[county_info['ID'].isin(counties)]
print("Filtered data for specified counties:")
filtered_counties[['County', 'Population', '%Poverty', 'PerCapitaIncome']]

Filtered data for specified counties:


County  Population   %Poverty  PerCapitaIncome
1040      Harlan County       27548  33.318182     16010.363636
2230     Malheur County       30421  24.414286     17966.428571
2241  Washington County      572071  10.446154     34970.817308
2968     Loudoun County      374558   3.884375     50391.015625

In [9]:
# Group by state and county to find the total population for each county
county_population = census_df.groupby(['State', 'County'])['TotalPop'].sum().reset_index()

# Finding the county with the highest population
highest_population_county = county_population.loc[county_population['TotalPop'].idxmax()]

# Finding the county with the lowest population
lowest_population_county = county_population.loc[county_population['TotalPop'].idxmin()]

print("County with the highest population:")
print(highest_population_county)

print("\nCounty with the lowest population:")
print(lowest_population_county)

County with the highest population:
State               California
County      Los Angeles County
TotalPop              10105722
Name: 204, dtype: object

County with the lowest population:
State               Texas
County      Loving County
TotalPop               74
Name: 2751, dtype: object


As we can see from the dictionary above:

* **Los Angeles County** has the highest population with 10105722
* **Loving County** has the lowest population with 74.

# C. Transform the COVID Data

1. Parse the `date` column in `covid_df` to datetime objects.
2. Add a column for the year and month.
3. Group by the new year-month column and `fips` (or a combination of `state` and `county`), and sum the cases and deaths over each month. Since the data seems to be cumulative, you'll want the last day of each month for each county.
4. Create the `county_id` column which matches the `fips`.
5. Finally, construct the table with the columns ["County", "Month", "#cases", "#deaths"] for the specific counties and months you're interested in.

In [10]:
# Convert the 'date' column to datetime
covid_df['date'] = pd.to_datetime(covid_df['date'])

# Create year-month column for grouping by month
covid_df['year_month'] = covid_df['date'].dt.to_period('M')

# Get last day entries for each month for each county which represent the cumulative number for the month
covid_monthly = covid_df.sort_values(['fips', 'date']).groupby(['year_month', 'fips']).last().reset_index()

covid_monthly

year_month     fips       date       county           state  cases  \
0        2020-01   4013.0 2020-01-31     Maricopa         Arizona      1   
1        2020-01   6037.0 2020-01-31  Los Angeles      California      1   
2        2020-01   6059.0 2020-01-31       Orange      California      1   
3        2020-01   6085.0 2020-01-31  Santa Clara      California      1   
4        2020-01  17031.0 2020-01-31         Cook        Illinois      2   
...          ...      ...        ...          ...             ...    ...   
36827    2021-02  72151.0 2021-02-20      Yabucoa     Puerto Rico    992   
36828    2021-02  72153.0 2021-02-20        Yauco     Puerto Rico    993   
36829    2021-02  78010.0 2021-02-20    St. Croix  Virgin Islands   1053   
36830    2021-02  78020.0 2021-02-20     St. John  Virgin Islands    212   
36831    2021-02  78030.0 2021-02-20   St. Thomas  Virgin Islands   1310   

       deaths  
0         0.0  
1         0.0  
2         0.0  
3         0.0  
4         0.0  
...       ...  
36827     NaN  
36828     NaN  
36829     9.0  
36830     1.0  
36831    15.0  

[36832 rows x 7 columns]

In [11]:
# Drop unnecessary columns
covid_monthly = covid_monthly[['fips', 'year_month', 'cases', 'deaths']]

# Rename the year_month back to the 'date' and 'fips' to 'county_id'
covid_monthly = covid_monthly.rename(columns={'year_month': 'Month', 'fips': 'county_id'})

# Add a county name column from covid_df to covid_monthly
covid_monthly = covid_monthly.merge(covid_df[['fips', 'county', 'state']].drop_duplicates(), left_on='county_id', right_on='fips', how='left')
covid_monthly['County'] = covid_monthly['county'] + ', ' + covid_monthly['state']

# Drop the additional 'fips', 'county', and 'state' columns
covid_monthly = covid_monthly.drop(['fips', 'county', 'state'], axis=1)

# Reorder columns
covid_monthly = covid_monthly[['County', 'county_id', 'Month', 'cases', 'deaths']]

# Convert the Month period to string format
covid_monthly['Month'] = covid_monthly['Month'].dt.strftime('%B %Y').astype(str)

covid_monthly.head()

County  county_id         Month  cases  deaths
0        Maricopa, Arizona     4013.0  January 2020      1     0.0
1  Los Angeles, California     6037.0  January 2020      1     0.0
2       Orange, California     6059.0  January 2020      1     0.0
3  Santa Clara, California     6085.0  January 2020      1     0.0
4           Cook, Illinois    17031.0  January 2020      2     0.0

In [12]:
counties = ["Malheur, Oregon"]
months = ["August 2020", "January 2021", "February 2021"]

# Now find the required data for the specified counties and months
specified_data = covid_monthly[
    (covid_monthly['County'].isin(counties)) & (covid_monthly['Month'].isin(months))
][['County', 'Month', 'cases', 'deaths']]

specified_data

County          Month  cases  deaths
16546  Malheur, Oregon    August 2020   1179    19.0
32619  Malheur, Oregon   January 2021   3259    56.0
35837  Malheur, Oregon  February 2021   3331    58.0

# D. Integrate COVID Data with ACS Data

To do this:

1. Transform the COVID Data
2. Fill the Required Table for Specific Counties

In [13]:
# Group by 'county' and 'state' and sum up the 'cases' and 'deaths'
covid_summary = covid_df.groupby(['county', 'state'], as_index=False).agg({
    'cases': 'sum',
    'deaths': 'sum'
})

# Cleanup: renaming for consistency with request
covid_summary.rename(columns={'county': 'County', 'state': 'State'}, inplace=True)

# For COVID data, concatenate County and State to form a unique identifier
covid_summary['ID'] = covid_summary['County'] + ', ' + covid_summary['State']
covid_summary['ID'] = covid_summary['ID'].str.replace(' County', '').str.strip()

# Do the same for the census DataFrame
census_df['ID'] = census_df['County'] + ', ' + census_df['State']

# Group by 'State' and 'County' and calculate the necessary statistics
census_summary = census_df.groupby(['State', 'County']).agg({
    'TotalPop': 'sum',
    'Poverty': 'mean',
    'Income': 'mean',
    'IncomePerCap': 'mean',
    'Employed': 'sum',
    'Unemployment': 'mean',
    'White': 'mean',
    'Black': 'mean',
    'Hispanic': 'mean',
    'Asian': 'mean',
    'Drive': 'mean',
    'Carpool': 'mean',
    'Transit': 'mean',
    'Walk': 'mean',
    'OtherTransp': 'mean',
    'WorkAtHome': 'mean',
    'MeanCommute': 'mean',
}).reset_index()

census_summary.columns = [
    'State', 'County', 'TotalPopulation', 'AveragePoverty', 'AverageIncome',
    'AverageIncomePerCap', 'TotalEmployed', 'AverageUnemployment', 'AverageWhite',
    'AverageBlack', 'AverageHispanic', 'AverageAsian', 'AverageDrive',
    'AverageCarpool', 'AverageTransit', 'AverageWalk', 'AverageOtherTransp',
    'AverageWorkAtHome', 'AverageMeanCommute'
]

# Cleanup: renaming for consistency with request
census_summary.rename(columns={'TotalPopulation': 'Population', 'AveragePoverty': 'Poverty%', 'AverageIncomePerCap': 'PerCapitaIncome'}, inplace=True)

# Add the unique identifier
census_summary['ID'] = census_summary['County'] + ', ' + census_summary['State']
census_summary['ID'] = census_summary['ID'].str.replace(' County', '').str.strip()

census_summary = census_summary.drop(['County', 'State'], axis=1)

# Merge covid_summary with census_summary on 'ID'
# COVID_summary = pd.merge(covid_summary, census_summary[['ID', 'Population', 'Poverty%', 'PerCapitaIncome']], on='ID', how='inner')
COVID_summary = pd.merge(covid_summary, census_summary, on='ID', how='inner')

# Calculate per 100K metrics
COVID_summary['TotalCasesPer100K'] = COVID_summary['cases'] * 100000 / COVID_summary['Population']
COVID_summary['TotalDeathsPer100K'] = COVID_summary['deaths'] * 100000 / COVID_summary['Population']

COVID_summary.head()

County           State    cases   deaths                         ID  \
0  Abbeville  South Carolina   206727   3934.0  Abbeville, South Carolina   
1   Accomack        Virginia   389637   5758.0         Accomack, Virginia   
2        Ada           Idaho  5230256  50724.0                 Ada, Idaho   
3      Adair            Iowa    70396   1792.0                Adair, Iowa   
4      Adair        Kentucky   178833   8026.0            Adair, Kentucky   

   Population   Poverty%  AverageIncome  PerCapitaIncome  TotalEmployed  \
0       24788  22.183333   35734.333333     19402.833333           9505   
1       32840  20.112500   41533.875000     24769.750000          13837   
2      435117  12.608475   62194.559322     32672.661017         214984   
3        7192   9.566667   50416.000000     28840.666667           3680   
4       19304  22.328571   36128.142857     18252.285714           7988   

   AverageUnemployment  AverageWhite  AverageBlack  AverageHispanic  \
0             9.383333     70.483333     25.800000         1.300000   
1             5.450000     63.525000     25.762500         7.837500   
2             4.633898     84.483051      1.332203         8.172881   
3             2.933333     96.633333      0.266667         1.800000   
4             6.828571     94.671429      2.414286         1.685714   

   AverageAsian  AverageDrive  AverageCarpool  AverageTransit  AverageWalk  \
0      0.283333     79.933333       10.716667        0.450000     1.616667   
1      0.925000     79.200000       10.812500        0.537500     2.662500   
2      2.608475     80.110169        7.642373        0.494915     1.871186   
3      0.333333     77.366667       12.200000        0.366667     2.800000   
4      0.071429     84.971429        9.214286        0.000000     2.228571   

   AverageOtherTransp  AverageWorkAtHome  AverageMeanCommute  \
0            2.083333           5.266667           26.450000   
1            2.025000           4.787500           22.000000   
2            3.203390           6.677966           19.642373   
3            0.466667           6.833333           22.266667   
4            0.571429           2.971429           23.685714   

   TotalCasesPer100K  TotalDeathsPer100K  
0       8.339802e+05        15870.582540  
1       1.186471e+06        17533.495737  
2       1.202034e+06        11657.554175  
3       9.788098e+05        24916.573971  
4       9.264039e+05        41576.875259

In [14]:
COVID_summary.head().to_csv()

',County,State,cases,deaths,ID,Population,Poverty%,AverageIncome,PerCapitaIncome,TotalEmployed,AverageUnemployment,AverageWhite,AverageBlack,AverageHispanic,AverageAsian,AverageDrive,AverageCarpool,AverageTransit,AverageWalk,AverageOtherTransp,AverageWorkAtHome,AverageMeanCommute,TotalCasesPer100K,TotalDeathsPer100K\n0,Abbeville,South Carolina,206727,3934.0,"Abbeville, South Carolina",24788,22.183333333333334,35734.333333333336,19402.833333333332,9505,9.383333333333333,70.48333333333333,25.8,1.3,0.2833333333333333,79.93333333333334,10.716666666666667,0.45,1.6166666666666665,2.0833333333333335,5.266666666666667,26.45,833980.1516862998,15870.58253993868\n1,Accomack,Virginia,389637,5758.0,"Accomack, Virginia",32840,20.1125,41533.875,24769.75,13837,5.45,63.525,25.762500000000003,7.8375,0.925,79.2,10.8125,0.5375,2.6625,2.025,4.7875,22.0,1186470.7673568819,17533.49573690621\n2,Ada,Idaho,5230256,50724.0,"Ada, Idaho",435117,12.608474576271187,62194.5593220339,32672.661016949154,214984,4.633898

In [15]:
# Define the counties of interest
counties_of_interest = [
    "Washington, Oregon",
    "Malheur, Oregon",
    "Loudoun, Virginia",
    "Harlan, Kentucky"
]

# Filter COVID_summary for those counties
counties_data = COVID_summary[COVID_summary['ID'].isin(counties_of_interest)]

In [16]:
# Select the relevant columns and explicitly create a copy
table_for_specific_counties = counties_data[['ID', 'Poverty%', 'TotalCasesPer100K']].copy()

# Show the required data table
table_for_specific_counties.rename(columns={'ID': 'County'}, inplace=True)

In [17]:
table_for_specific_counties

County   Poverty%  TotalCasesPer100K
1176    Harlan, Kentucky  33.318182       7.477276e+05
1677   Loudoun, Virginia   3.884375       6.665056e+05
1737     Malheur, Oregon  24.414286       1.491187e+06
2903  Washington, Oregon  10.446154       3.771104e+05

# E. Analysis

In [18]:
# First, we filter the data for Oregon counties only.
oregon_data = COVID_summary[COVID_summary['ID'].str.contains(", Oregon")]

In [19]:
# COVID total cases vs. % population in poverty (for Oregon counties)
R1 = oregon_data['TotalCasesPer100K'].corr(oregon_data['Poverty%'])

# COVID total deaths vs. % population in poverty (for Oregon counties)
R2 = oregon_data['TotalDeathsPer100K'].corr(oregon_data['Poverty%'])

# COVID total cases vs. Per Capita Income level (for Oregon counties)
R3 = oregon_data['TotalCasesPer100K'].corr(oregon_data['PerCapitaIncome'])

# COVID total deaths vs. Per Capita Income level (for Oregon counties)
R4 = oregon_data['TotalDeathsPer100K'].corr(oregon_data['PerCapitaIncome'])

# COVID total cases vs. % population in poverty (for all USA counties)
R5 = COVID_summary['TotalCasesPer100K'].corr(COVID_summary['Poverty%'])

# COVID total deaths vs. % population in poverty (for all USA counties)
R6 = COVID_summary['TotalDeathsPer100K'].corr(COVID_summary['Poverty%'])

# COVID total cases vs. Per Capita Income level (for all USA counties)
R7 = COVID_summary['TotalCasesPer100K'].corr(COVID_summary['PerCapitaIncome'])

# COVID total deaths vs. Per Capita Income level (for all USA counties)
R8 = COVID_summary['TotalDeathsPer100K'].corr(COVID_summary['PerCapitaIncome'])

#COVID total cases vs. Average Poverty Level (all counties)
R9 = COVID_summary['cases'].corr(COVID_summary['Poverty%'])

#COVID total deaths vs. Average Income (all counties)
R10 = COVID_summary['deaths'].corr(COVID_summary['AverageIncome'])

#COVID total cases vs. Average Unemployment Rate (all counties)
R11 = COVID_summary['cases'].corr(COVID_summary['AverageUnemployment'])

#COVID total deaths vs. Average Income Per Capita (all counties)
R12 = COVID_summary['deaths'].corr(COVID_summary['PerCapitaIncome'])

#COVID total cases vs. Average Commute Time (all counties)
R13 = COVID_summary['cases'].corr(COVID_summary['AverageMeanCommute'])

#COVID total cases vs. Average White Population Percentage (all counties)
R14 = COVID_summary['cases'].corr(COVID_summary['AverageWhite'])

#COVID total cases vs. Average Hispanic Population Percentage (all counties)
R15 = COVID_summary['cases'].corr(COVID_summary['AverageHispanic'])

#COVID total cases vs. Average Black Population Percentage (all counties)
R16 = COVID_summary['cases'].corr(COVID_summary['AverageBlack'])

#COVID total deaths vs. Average Walking Commute Percentage (all counties)
R17 = COVID_summary['deaths'].corr(COVID_summary['AverageWalk'])

#COVID total deaths vs. Average Walking Commute Percentage (Oregon counties)
R18 = COVID_summary['deaths'].corr(oregon_data['AverageWalk'])

#COVID total cases vs. Average Work From Home Percentage (all counties)
R19 = COVID_summary['cases'].corr(COVID_summary['AverageWorkAtHome'])

#COVID total cases vs. Average Work From Home Percentage (Oregon counties)
R20 = COVID_summary['cases'].corr(oregon_data['AverageWorkAtHome'])

In [20]:
import pandas as pd

columns = ["Description", "R value"]
data = [
    ["For Oregon Counties only: correlation between % poverty and COVID cases", R1],
    ["For Oregon Counties only: correlation between % poverty and COVID deaths", R2],
    ["For Oregon counties only: correlation between PerCapitaIncome and COVID cases", R3],
    ["For Oregon counties only: correlation between PerCapitaIncome and COVID deaths", R4],
    ["For all counties: correlation between % poverty and COVID cases", R5],
    ["For all counties: correlation between % poverty and COVID deaths", R6],
    ["For all counties: correlation between PerCapitaIncome and COVID cases", R7],
    ["For all counties: correlation between PerCapitaIncome and COVID deaths", R8],
    ["For all counties: correlation between cases and Average Poverty", R9],
    ["For all counties: correlation between deaths and Average Income", R10],
    ["For all counties: correlation between cases and Average Unemployment", R11],
    ["For all counties: correlation between deaths and Average Income Per Cap", R12],
    ["For all counties: correlation between cases and Average Commute Time", R13],
    ["For all counties: correlation between cases and Average White Pop %", R14],
    ["For all counties: correlation between cases and Average Hispanic Pop %", R15],
    ["For all counties: correlation between cases and Average Black Pop %", R16],
    ["For all counties: correlation between deaths and walking", R17],
    ["For Oregon counties only: correlation between deaths and walking", R18],
    ["For all counties: correlation between cases and Working From Home", R19],
    ["For Oregon counties only: correlation between cases and Working From Home", R20]
]

table = pd.DataFrame(data, columns=columns)
table

Description  \
0          For Oregon Counties only: correlation between % poverty and COVID cases   
1         For Oregon Counties only: correlation between % poverty and COVID deaths   
2    For Oregon counties only: correlation between PerCapitaIncome and COVID cases   
3   For Oregon counties only: correlation between PerCapitaIncome and COVID deaths   
4                  For all counties: correlation between % poverty and COVID cases   
5                 For all counties: correlation between % poverty and COVID deaths   
6            For all counties: correlation between PerCapitaIncome and COVID cases   
7           For all counties: correlation between PerCapitaIncome and COVID deaths   
8                  For all counties: correlation between cases and Average Poverty   
9                  For all counties: correlation between deaths and Average Income   
10            For all counties: correlation between cases and Average Unemployment   
11         For all counties: correlation between deaths and Average Income Per Cap   
12            For all counties: correlation between cases and Average Commute Time   
13             For all counties: correlation between cases and Average White Pop %   
14          For all counties: correlation between cases and Average Hispanic Pop %   
15             For all counties: correlation between cases and Average Black Pop %   
16                        For all counties: correlation between deaths and walking   
17                For Oregon counties only: correlation between deaths and walking   
18               For all counties: correlation between cases and Working From Home   
19       For Oregon counties only: correlation between cases and Working From Home   

     R value  
0   0.203004  
1   0.291566  
2  -0.322050  
3  -0.411254  
4   0.185550  
5   0.258117  
6  -0.203981  
7  -0.168988  
8  -0.015058  
9   0.239131  
10  0.047053  
11  0.216175  
12  0.113611  
13 -0.235238  
14  0.204420  
15  0.086404  
16 -0.033752  
17 -0.231274  
18 -0.025949  
19 -0.221193

In [22]:
# Now let's filter for |R| > 0.5
strong_corrs = table[table['R value'].abs() > 0.5]

# Plotting
for index, row in strong_corrs.iterrows():
    plt.figure(figsize=(10, 6))
    sns.scatterplot(data=row['Data'], x=row['X'], y=row['Y'], marker='o')
    plt.title(f"{row['Description']} (R = {row['R value']:.2f})")
    plt.xlabel(row['X'])
    plt.ylabel(row['Y'])
    plt.show()

I couldn't seem to find one correlation that was < -0.5 or > 0.5